### PR TITLE
chore: [#186134528] empty roadmap bug when discarding new business

### PR DESCRIPTION
**Pull request by somabadri**
## Description

Fix to rebuild roadmap if the steps are empty after discarding the process of starting a new business. 

### Ticket

[#186134528](https://www.pivotaltracker.com/story/show/186134528)

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

**Review by cbrunefearless — APPROVED** [review]
lgtm

File: web/src/lib/data-hooks/useRoadmap.ts [hunk 1]
```diff
@@ -31,8 +31,10 @@ export const useRoadmap = (): UseRoadmapReturnValue => {
     return [...new Set(sections)];
   }, [roadmap]);
 
+  const isRoadMapEmpty = roadmap?.steps.length === 0 || roadmap?.tasks.length === 0;
+
   useMountEffectWhenDefined(() => {
-    if (!roadmap) {
+    if (!roadmap || isRoadMapEmpty) {
```
Comment (seidior) [inline web/src/lib/data-hooks/useRoadmap.ts:37]: Given this is the only place where you use the variable `isRoadMapEmpty`, I'd love to see you combine all your conditionals into one:

```typescript
const roadmapNeedsRebuild = !roadmap || roadmap.steps.length === 0 || roadmap.tasks.length === 0;
```

(Plus, this allows you to drop the optionals in your conditional.)

That way, your branching statement reads a little better:

```typescript
if (roadmapNeedsRebuild) {
  buildAndSetRoadmap();
}
```

(I'm totally open to better names, btw.)

**Review by seidior — CHANGES_REQUESTED** [review]
Great work hunting this down, creating reproducible manual tests, creating a failing test, coming up with a hot fix, and validating your test. I look forward to the next round of your bug hunting to figure out why this situation was a problem in the first place!

File: web/src/lib/data-hooks/useRoadmap.ts (hunk 1 above)
Comment (jphechter) [reply web/src/lib/data-hooks/useRoadmap.ts:37]: @seidior Not sure if we've talked about this at all, but we've been trying to label our code comments with a weight. We've been using a [modified version of the rock scale](https://docs.google.com/document/d/1aTAQthQf704Ed0hOEDAdU7Vq41B_YrwraiiU05PugYY/edit#heading=h.njm8kqnslu7y) to do this. The idea is to help the reviewer consider and the reviewee to understand how important a given comment is.

Ex: [pebble] Suggestion to follow.

**Review by jphechter — COMMENTED** [review]
I like Spencer's suggestion. Otherwise, this looks good to me.

File: web/src/lib/data-hooks/useRoadmap.ts (hunk 1 above)
Comment (somabadri) [reply web/src/lib/data-hooks/useRoadmap.ts:37]: Going with `rebuildRoadmap` as the variable name.

**Pull request merged**